### DISCUSSION TITLE
Handle row not found for `.one`

### DISCUSSION THREAD

**Discussion by phungleson** [SeaQL/sea-orm#1109] [General]
Hey peeps,

From the web/api perspective find that when users request for a record by id and it is not found, we would like to display not found page (404)/error message.

Maybe `sea-orm` should also have 2 methods, `one` to return `Option` and `one_unwrapped` or `unwrap_one` to return `DbErr::RowNotFound`. As returning `DbErr::RowNotFound` makes error conversion more fluent in the context of Rust.

It would also be helpful to know other pre-art about how to handle this case.

Cheers,

**Comment by billy1624** [comment]
Hey @phungleson, I think returning an `Option<T>` for the `one` method make sense. And user can always map the `None` as `Err( ... )` with [`ok_or`](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or).

**Comment by phungleson** [comment]
Hey @billy1624 thanks!

Yeah I think `one` returning `Option` makes sense but in certain scenarios returning `DbErr::RowNotFound` also makes sense.

My suggestion is to keep `one` as it is and add another method where it returns `DbErr::RowNotFound`. So that we can convert `From<DbErr> for ServerError` and return `404`  smoothly using `?` instead of littering the code with  `ok_or`.

I am not sure how other framework works, but for example in Rails, `.find` will throw `RecordNotFound` where as `find_by_id` will return `nil`. It is up to the caller to decide depending on their needs.

**Reply by frederikhors** [reply]
I'm also interested in this. Can you PR, @phungleson?
